arch/riscv/mach-dpu/mach.c

```diff
@@ -80,7 +100,38 @@ static void dpu_boot_spi(void)
 
 #ifdef CONFIG_DPU_LOAD_SSI_FLASH
 #ifdef CONFIG_DPU_SIM_SSI_BACKDOOR
-#define dpu_load_ssi(boot_entry, boot_file)	do { } while (0)
```

Comment:
We use SIM_SSI_BACKDOOR to refer a behavior that the SSI data has already been loaded to the boot target address.

What if you define several new Kconfig options:
CONFIG_DPU_LOAD_ROM
CONFIG_DPU_LOAD_PCIE_MEM
And might be look like the followings:
`
choice
  prompt "Storage to load from"
  default DPU_LOAD_SSI_FLASH

config DPU_LOAD_SSI_FLASH
  bool "SSI flash"

config DPU_LOAD_ROM
  bool "ROM"

config DPU_LOAD_PCIE_MEM
  bool "PCIe memory"

endchoice
`

And you can define following functions:
`
#ifdef CONFIG_DPU_LOAD_ROM
static void dpu_load_rom(...)
{
  ...
}
#else
stub it
#endif

#ifdef CONFIG_DPU_LOAD_PCIE_MEM
static void dpu_load_pcie(...)
{
  ...
}
#else
stub it
#endif
`
This follows the stub rule: creates stub around the macro/function boundary rather than inside of the function.

Then in the caller side, you can alter the original caller as:
`
 dpu_load_ssi(...);
+dpu_load_rom(...);
+dpu_load_pcie(..);
`